rust/src/daemon.rs

```diff
@@ -246,6 +254,104 @@ pub(crate) fn deployment_layeredmeta_load(
     deployment_layeredmeta_from_commit(deployment.gobj_rewrap(), commit.gobj_rewrap())
 }
 
+#[context("Loading origin status")]
+pub(crate) fn variant_add_remote_status(
+    mut repo: Pin<&mut crate::FFIOstreeRepo>,
+    refspec: &str,
+    base_checksum: &str,
+    mut dict: Pin<&mut crate::FFIGVariantDict>,
+) -> CxxResult<()> {
+    let repo = &repo.gobj_wrap();
+    let dict = &dict.gobj_wrap();
+
+    let (ostree_remote, _) = ostree::parse_refspec(refspec)?;
+    let mut gpg_verify = false;
+    if let Some(ref ostree_remote) = ostree_remote {
```

Comment:
At the cost of additional indentation, this could be `let gpg_verify = if let Some(...) { }` because conceptually it's returning the boolean to the outer scope.  Then we wouldn't need `mut`.

But obviously fine as is!

Reply:
Actually now that I look, a bit better as:

```
let (remote, gpg_verify) = if let Some(...) { }
```

Then we could also drop the ` let ostree_remote = ostree_remote.unwrap();` hunk.

Reply:
Almost. We still need an `else` to that `if` that would have to duplicate the `gpg-verify` injection and return. Didn't feel as elegant, so changed it to `let gpg_verify = if ...`.